sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -926,7 +932,12 @@ case class ColumnStatsMap(originalMap: AttributeMap[ColumnStat]) {
         // no need to scale down since it is already down to 1 (for skewed distribution case)
         colStat.distinctCount
       }
-      attr -> colStat.copy(distinctCount = newNdv)
+      val newNullCount = if (colStat.nullCount.isEmpty) {
+        None
+      } else {
+        Some(colStat.nullCount.get.min(rowsAfterFilter))
```

Comment:
Shall we follow how we update ndv and consider the ratio? 
```
        // Update ndv based on the overall filter selectivity: scale down ndv if the number of rows
        // decreases; otherwise keep it unchanged.
        Some(EstimationUtils.updateNdv(oldNumRows = rowsBeforeFilter,
          newNumRows = rowsAfterFilter, oldNdv = oriColStat.distinctCount.get))
```